src/less/icon/base/icon.less

```diff
@@ -73,6 +74,7 @@ svg {
     }
 
     &.icon--checkbox-checked-large {
+        .color-token(checkbox-mixed-foreground);
```

Comment:
This should not be needed if the fill is applied on the SVG (I checked it locally)